### PR TITLE
[ROSBE] 2.2.0 Updates at it's finest

### PR DESCRIPTION
**Pull request by dreimer1986**
Update changelog last time
Removed old hack
Readded dlls needed for cmake
Add more rights to Users in Install root dir.
Semi fix GIT mess

### CODE REVIEW

**Comment by learn-more** [comment]
Your first commit (e9bf78b) is still with your nickname,
and by only having the component on the first line, the git history doesnt really read nicely:

**Comment by learn-more** [comment]
@dreimer1986 Any updates?

File: RosBE-Windows/RosBE.nsi [hunk 1]
```diff
@@ -169,6 +169,7 @@ Section -BaseFiles SEC01
     File /r Components\Bin\gdb.exe
     File /r Components\Bin\gdbserver.exe
     File /r Components\Bin\getdate.exe
+	File /r Components\Bin\libeay32.dll
```
Comment (binarymaster) [inline RosBE-Windows/RosBE.nsi]: Indentation problem here.

File: RosBE-Windows/RosBE-Windows_RosBE.nsi [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+
+    ;;
+    ;; Clean up the registry.
+    ;;
+    DeleteRegKey ${PRODUCT_UNINST_ROOT_KEY} "${PRODUCT_UNINST_KEY}"
+    DeleteRegKey HKCU "${PRODUCT_DIR_REGKEY}"
+    SetAutoClose true
+SectionEnd
+
+;; HACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACK
```
Comment (learn-more) [inline RosBE-Windows/RosBE-Windows_RosBE.nsi]: This might be a hack?

Comment (dreimer1986) [reply RosBE-Windows/RosBE-Windows_RosBE.nsi]: This stuff is outdated and will be replaced soon. I still hate GIT btw!

File: RosBE-Windows/RosBE.nsi [hunk 3]
```diff
@@ -183,8 +184,10 @@ Section -BaseFiles SEC01
     File /r Components\Bin\MSVCM90.dll
     File /r Components\Bin\msvcp60.dll
     File /r Components\Bin\MSVCP90.dll
+	File /r Components\Bin\MSVCP100.dll
```
Comment (HBelusca) [inline RosBE-Windows/RosBE.nsi]: Ditto

File: RosBE-Windows/RosBE.nsi [hunk 4]
```diff
@@ -183,8 +184,10 @@ Section -BaseFiles SEC01
...
     File /r Components\Bin\MSVCP120.dll
     File /r Components\Bin\MSVCR90.dll
+	File /r Components\Bin\MSVCR100.dll
```
Comment (HBelusca) [inline RosBE-Windows/RosBE.nsi]: Ditto

File: RosBE-Windows/RosBE.nsi [hunk 5]
```diff
@@ -197,8 +200,10 @@ Section -BaseFiles SEC01
     File /r Components\Bin\rquote.exe
     File /r Components\Bin\scut.exe
     File /r Components\Bin\sdiff.exe
+	File /r Components\Bin\ssleay32.dll
```
Comment (HBelusca) [inline RosBE-Windows/RosBE.nsi]: Ditto

File: RosBE-Windows/RosBE.nsi [hunk 6]
```diff
@@ -197,8 +200,10 @@ Section -BaseFiles SEC01
...
     File /r Components\Bin\tee.exe
     File /r Components\Bin\wget.exe
+	File /r Components\Bin\zlib1.dll
```
Comment (HBelusca) [inline RosBE-Windows/RosBE.nsi]: Ditto

File: RosBE-Windows/RosBE.nsi [hunk 7]
```diff
@@ -275,20 +308,25 @@ Section /o "PowerShell Version" SEC05
...
-    IfFileExists "$SMPROGRAMS\$ICONS_GROUP\ReactOS Build Environment ${PRODUCT_VERSION}.lnk" +13 0
+    IfFileExists "$SMPROGRAMS\$ICONS_GROUP\ReactOS Build Environment ${PRODUCT_VERSION}.lnk" +18 0
         !insertmacro MUI_STARTMENU_WRITE_BEGIN Application
             CreateDirectory "$SMPROGRAMS\$ICONS_GROUP"
             SetOutPath $REACTOS_SOURCE_DIRECTORY
             IfFileExists "$INSTDIR\RosBE.cmd" 0 +2
                 CreateShortCut "$SMPROGRAMS\$ICONS_GROUP\ReactOS Build Environment ${PRODUCT_VERSION}.lnk" "$SYSDIR\cmd.exe" '/t:0A /k "$INSTDIR\RosBE.cmd"' "$INSTDIR\rosbe.ico"
             IfFileExists "$INSTDIR\RosBE.ps1" 0 +2
                 CreateShortCut "$SMPROGRAMS\$ICONS_GROUP\ReactOS Build Environment ${PRODUCT_VERSION} - PS.lnk" "$SYSDIR\WindowsPowerShell\v1.0\powershell.exe" "-noexit &'$INSTDIR\RosBE.ps1'" "$INSTDIR\rosbe.ico"
+			IfFileExists "$INSTDIR\amd64\*" 0 +6
```
Comment (HBelusca) [inline RosBE-Windows/RosBE.nsi]: Are there indentation problems in the following lines?

File: RosBE-Windows/RosBE.nsi [hunk 8]
```diff
@@ -299,35 +337,45 @@ Section -StartMenuShortcuts SEC06
...
     ;; Add our desktop shortcuts.
     ;;
-    IfFileExists "$DESKTOP\ReactOS Build Environment ${PRODUCT_VERSION}.lnk" +6 0
+    IfFileExists "$DESKTOP\ReactOS Build Environment ${PRODUCT_VERSION}.lnk" +12 0
         SetOutPath $REACTOS_SOURCE_DIRECTORY
         IfFileExists "$INSTDIR\RosBE.cmd" 0 +2
             CreateShortCut "$DESKTOP\ReactOS Build Environment ${PRODUCT_VERSION}.lnk" "$SYSDIR\cmd.exe" '/t:0A /k "$INSTDIR\RosBE.cmd"' "$INSTDIR\rosbe.ico"
         IfFileExists "$INSTDIR\RosBE.ps1" 0 +2
             CreateShortCut "$DESKTOP\ReactOS Build Environment ${PRODUCT_VERSION} - PS.lnk" "$SYSDIR\WindowsPowerShell\v1.0\powershell.exe" "-noexit &'$INSTDIR\RosBE.ps1'" "$INSTDIR\rosbe.ico"
+		IfFileExists "$INSTDIR\amd64\*" 0 +6
```
Comment (HBelusca) [inline RosBE-Windows/RosBE.nsi]: Are there indentation problems in the following lines?

File: RosBE-Windows/RosBE.nsi [hunk 9]
```diff
@@ -358,6 +406,10 @@ Function un.onInit
         Delete /REBOOTOK "$QUICKLAUNCH\ReactOS Build Environment ${PRODUCT_VERSION}.lnk"
         Delete /REBOOTOK "$DESKTOP\ReactOS Build Environment ${PRODUCT_VERSION} - PS.lnk"
         Delete /REBOOTOK "$QUICKLAUNCH\ReactOS Build Environment ${PRODUCT_VERSION} - PS.lnk"
+		Delete /REBOOTOK "$DESKTOP\ReactOS Build Environment ${PRODUCT_VERSION} AMD64.lnk"
```
Comment (HBelusca) [inline RosBE-Windows/RosBE.nsi]: Indentation problem.

File: RosBE-Windows/RosBE.nsi [hunk 10]
```diff
@@ -299,35 +337,45 @@ Section -StartMenuShortcuts SEC06
...
     ;; Add our quick launch shortcuts.
     ;;
-    IfFileExists "$QUICKLAUNCH\ReactOS Build Environment ${PRODUCT_VERSION}.lnk" +6 0
+    IfFileExists "$QUICKLAUNCH\ReactOS Build Environment ${PRODUCT_VERSION}.lnk" +12 0
         SetOutPath $REACTOS_SOURCE_DIRECTORY
         IfFileExists "$INSTDIR\RosBE.cmd" 0 +2
             CreateShortCut "$QUICKLAUNCH\ReactOS Build Environment ${PRODUCT_VERSION}.lnk" "$SYSDIR\cmd.exe" '/t:0A /k "$INSTDIR\RosBE.cmd"' "$INSTDIR\rosbe.ico"
         IfFileExists "$INSTDIR\RosBE.ps1" 0 +2
             CreateShortCut "$QUICKLAUNCH\ReactOS Build Environment ${PRODUCT_VERSION} - PS.lnk" "$SYSDIR\WindowsPowerShell\v1.0\powershell.exe" "-noexit &'$INSTDIR\RosBE.ps1'" "$INSTDIR\rosbe.ico"
+		IfFileExists "$INSTDIR\amd64\*" 0 +6
```
Comment (HBelusca) [inline RosBE-Windows/RosBE.nsi]: Are there indentation problems in the following lines?

**Comment by HBelusca** [comment]
Does this new version of RosBE work on ReactOS?

**Comment by dreimer1986** [comment]
VirtualBox_ROS_28_04_2019_01_26_43

As long as you ignore wget causing trouble in ws2_32 thx to being made for anyting but Windows XP.... Looks good for now. And if you wanna know why: https://eternallybored.org/misc/wget/ XP support dropped...

File: RosBE-Windows/Powershell/RosBE.ps1 [hunk 11]
```diff
@@ -69,7 +69,7 @@ $global:_BUILDBOT_SVNSKIPMAINTRUNK = "0"
 $ENV:CCACHE_SLOPPINESS = "time_macros"
 
 # Fix Bison package path (just in case RosBE is installed in a path which contains spaces)
-$ENV:BISON_PKGDATADIR = ((New-Object -ComObject Scripting.FileSystemObject).GetFolder("$_ROSBE_HOST_MINGWPATH\share\bison")).ShortPath
+$ENV:BISON_PKGDATADIR = ((New-Object -ComObject Scripting.FileSystemObject).GetFolder("$_ROSBE_BASEDIR\bin\data")).ShortPath
```
Comment (ColinFinck) [inline RosBE-Windows/Powershell/RosBE.ps1]: With my Bison 3.5 binary from https://svn.reactos.org/RosBE-Temp/RosBE-2.2_Dec2019.7z, this environment variable is required and the data path is share\bison.
We also need another environment variable "M4" with the full path to the m4.exe binary (which is also shipped in a defined version in RosBE-2.2_Dec2019.7z).

File: RosBE-Windows/Root/ChangeLog.txt [hunk 12]
```diff
@@ -1,5 +1,23 @@
-*** Feb 01th, 2018 - RosBE 2.1.6 Released
-
+*** Aug 17th, 2019 - RosBE 2.2.0 Released
+
+- Updated GCC to 8.3.0 (Colin Finck, Daniel Reimer)
+  - i686-w64-mingw32 AND x86_64-w64_mingw32 (optional) in one setup!
+- Updated binutils to 2.32 (Colin Finck, Daniel Reimer)
+- Updated mingw-w64 to 6.0.0 (Colin Finck, Daniel Reimer)
+- Updated Flex to 2.6.4 (Daniel Reimer) (https://sourceforge.net/projects/winflexbison/)
```
Comment (ColinFinck) [inline RosBE-Windows/Root/ChangeLog.txt]: It's Bison 3.5 and Flex 2.6.4+ (revision 8b1fbf674f2e038df9cf1fe7725617e3837ae2a9) with my updated binaries from https://svn.reactos.org/RosBE-Temp/RosBE-2.2_Dec2019.7z
We are using the very same versions for RosBE-Unix now and can therefore safely make Bison and Flex a build dependency.

The winflexbison sources are heavily modified to an extent, where they don't build on anything else than Windows/MSVC anymore and many code paths needed to be disabled for this.
Therefore, winflexbison wasn't suitable for sourcing reliable versions that can be shared between RosBE-Windows and RosBE-Unix

File: RosBE-Windows/Root/RosBE.cmd [hunk 13]
```diff
@@ -90,7 +90,7 @@ set _BUILDBOT_SVNSKIPMAINTRUNK=0
 set CCACHE_SLOPPINESS=time_macros
 
 :: Fix Bison package path (just in case RosBE is installed in a path which contains spaces)
-set BISON_PKGDATADIR=%~ds0%~sp0%i386\share\bison
+set BISON_PKGDATADIR=%~ds0%~sp0%bin\data
```
Comment (ColinFinck) [inline RosBE-Windows/Root/RosBE.cmd]: Requires same change as above

**Review by ColinFinck — CHANGES_REQUESTED** [review]

**Comment by HBelusca** [comment]
Do the RosBE 2.2.0 components work on ReactOS?

File: RosBE-Windows/Powershell/RosBE.ps1 (hunk 11 above)
Comment (dreimer1986) [reply RosBE-Windows/Powershell/RosBE.ps1]: Fixed I hope.

File: RosBE-Windows/Root/ChangeLog.txt (hunk 12 above)
Comment (dreimer1986) [reply RosBE-Windows/Root/ChangeLog.txt]: Fixed I hope.

File: RosBE-Windows/Root/RosBE.cmd (hunk 13 above)
Comment (dreimer1986) [reply RosBE-Windows/Root/RosBE.cmd]: Fixed I hope.

**Comment by dreimer1986** [comment]
https://dreimer.eu/rosbe/RosBE-2.2.0.exe Test it!

**Comment by Zero3K** [comment]
Its stating that the CMake at /RosBE/bin isn't the custom one even though it is (since it has a date of 12/26/2019).

**Comment by dreimer1986** [comment]
> Its stating that the CMake at /RosBE/bin isn't the custom one even though it is (since it has a date of 12/26/2019).

Known. This check will be removed

**Pull request merged**